lib/internal/http2/core.js

```diff
@@ -2286,15 +2262,34 @@ function sessionOnSocketError(error, socket) {
     this[kServer].emit('socketError', error, socket, this);
 }
 
+// When the session times out on the server, attempt a graceful shutdown
+function sessionOnTimeout() {
+  debug('session timeout');
+  process.nextTick(() => {
+    // if destroyed or destryoing, do nothing
+    if (this.destroyed || this.destroying)
```

Comment:
should be faster to use the `kState` references here instead of the getters

Reply:
Posting an updated version shortly. Thanks for the review!